tests/data_generator/datagenerator.go

```diff
@@ -0,0 +1,300 @@
+	}
+}
+
+func AddConcepts(concepts []Concept) {
+	log.Printf("Processing %d concepts...", len(concepts))
+	nextConceptId := tests.GetLastConceptId(sourceId) + 1
+	for _, concept := range concepts {
+		var nrClones = 1
+		if concept.CloneCount > 0 {
+			nrClones = concept.CloneCount
+			log.Printf("** Clone directive found! Cloning this concept %d times **", concept.CloneCount)
+		}
+		for i := 0; i < nrClones; i++ {
+			AddConceptAndMaybeAddObservations(nextConceptId, concept)
+			nextConceptId++
+		}
+	}
+}
+
+func AddConceptAndMaybeAddObservations(nextConceptId int64, concept Concept) {
+	var conceptId = nextConceptId
+	var conceptName = concept.ConceptName
+	var conceptClassId = "MVP Continuous"
+	if concept.Id != "" {
+		conceptId = utils.ParseInt64(concept.Id)
+	}
+	if concept.ConceptValueName != "" {
+		conceptName = concept.ConceptValueName
+		// TODO - validate there is only one:
+		conceptId = utils.ParseInt64(concept.PossibleValues[0])
+	}
+	if concept.ValueType == "concept" {
+		conceptClassId = "MVP Ordinal"
+	}
+	// If still empty, generate a name:
+	if conceptName == "" {
+		conceptName = fmt.Sprintf("%s-%d", concept.Concept, conceptId)
+	}
+	// If concept id was already added before, skip inserting it:
+	if utils.Pos(conceptId, conceptIds) == -1 {
+		tests.ExecSQLStringOrFail(
+			fmt.Sprintf(
+				"INSERT into %s.concept "+
+					"(concept_id,concept_code,concept_name,domain_id,concept_class_id,standard_concept,valid_start_date,valid_end_date,invalid_reason) "+
+					"values "+
+					"(%d,'%s','%s','%s','%s','%s','%s','%s',NULL)",
+				tests.GetOmopDataSourceForSourceId(sourceId).Schema,
+				conceptId, concept.Concept, conceptName, "Person", conceptClassId, "S", "1970-01-01", "2097-12-31"),
+			sourceId)
+		// keep track of added concepts, so we don't add them twice:
+		conceptIds = append(conceptIds, conceptId)
+	}
+	// Check if there are also personIds:
+	var personIds = concept.PersonIdsOverride
+	if personIds == nil {
+		personIds = concept.PersonIds
+	}
+	// If there are any personIds, add also entries linking the persons and the concept in `observation` table:
+	for _, personId := range personIds {
+		AddObservationForPerson(conceptId, concept, personId)
+	}
+}
+
+func AddObservationForPerson(conceptId int64, concept Concept, personId int64) {
+	AddPerson(personId)
+	var valueAsNumber = "NULL"
+	var valueAsConceptId = "NULL"
+	if concept.ValueType == "number" {
+		if concept.PossibleValues == nil {
+			valueAsNumber = strconv.Itoa(rand.Intn(10000))
+		}
+		if len(concept.PossibleValues) > 0 {
+			max := len(concept.PossibleValues)
+			randIndex := 0
+			if max > 1 {
+				randIndex = rand.Intn(max - 1)
+			}
+			valueAsNumber = concept.PossibleValues[randIndex]
+		}
+	} else if concept.ValueType == "concept" && len(concept.PossibleValues) > 0 {
+		max := len(concept.PossibleValues)
+		randIndex := 0
+		if max > 1 {
+			randIndex = rand.Intn(max - 1)
+		}
+		valueAsConceptId = concept.PossibleValues[randIndex]
```

Comment:
would you need to know what to expect here? How could you validate counts in a test without knowing?